tests/aws/services/stepfunctions/templates/scenarios/statemachines/map_state_config_distributed_reentrant_lambda.json5

```diff
@@ -0,0 +1,64 @@
+{
+  "Comment": "MAP_STATE_CONFIG_DISTRIBUTED_REENTRANT_LAMBDA",
+  "StartAt": "StartState",
+  "States": {
+    // Populate memory with two fields: number of iterations left, and input values.
+    "StartState": {
+      "Type": "Pass",
+      "Parameters": {
+        "Iterations": 2,
+        "Values": [
+          "HelloWorld"
+        ]
+      },
+      "Next": "BeforeIteration"
+    },
+    // Prepare the iteration by updating the iterations count.
+    "BeforeIteration": {
+      "Type": "Pass",
+      "Parameters": {
+        "Iterations.$": "States.MathAdd($.Iterations, -1)",
+        "Values.$": "$.Values"
+      },
+      "Next": "IterationBody"
+    },
+    // Run a distributed map state on the values field.
+    "IterationBody": {
+      "Type": "Map",
+      "MaxConcurrency": 1,
+      "ItemsPath": "$.Values",
+      "ItemProcessor": {
+        "ProcessorConfig": {
+          "Mode": "DISTRIBUTED",
+          "ExecutionType": "STANDARD"
+        },
+        "StartAt": "ProcessValue",
+        "States": {
+          "ProcessValue": {
+            "Type": "Task",
+            "Resource": "_tbd_",
```

Comment:
That seems to be the main difference here, having the Lambda resource invocation. Consider highlighting the important pieces through comments such that it is easier to spot them within lengthy state machine definitions.